staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/validation/validation.go

```diff
@@ -242,3 +242,44 @@ func convertJSONSchemaPropsOrStringArray(in *apiextensions.JSONSchemaPropsOrStri
 	}
 	return nil
 }
+
+// RemoveUnknownFields removes those fields from the object which are not specified in the properties construct of the schema.
+func RemoveUnknownFields(object map[string]interface{}, schema *apiextensions.JSONSchemaProps) {
+	if object == nil || schema == nil || schema.Properties == nil {
+		return
+	}
+
+	for field := range object {
+		// don't consider typemeta and objectmeta
+		if field == "apiVersion" || field == "kind" || field == "metadata" {
```

Comment:
We should probably introduce default validation for these fields.

Reply:
We have that already:

https://github.com/kubernetes/kubernetes/blob/11ecad262939c72404b592e721f172f1dac352dd/staging/src/k8s.io/apiextensions-apiserver/pkg/registry/customresource/strategy.go#L125-L139

Reply:
Thanks for the link.  Is the behavior for these fields are validated documented in the user facing CRD docs?  Which version of the metadata schema does this validate against and how do we know that version matches the version in what the user defined controllers expect it to be?